### DISCUSSION TITLE
Development workflow newbie question about modules

### DISCUSSION THREAD

**Discussion by juanmaia** [pyscaffold/pyscaffold#549] [Q&A] [answered]
Hi folks, I have a very simple question. After creating a project with PyScaffold, let's say I want to modify the `fib` function and test it running the file in the command line.

After I run `$ python src/myproject/skeleton.py fib 10` I get the following error:
```
Traceback (most recent call last):
  File "src/myproject/skeleton.py", line 27, in <module>
    from myproject import __version__
ModuleNotFoundError: No module named 'myproject'
```

Only after I install with `pip install .` it will work, but then everything imported with `from myproject import <module_here>` will be cached inside site-packages and won't reflect if I change the file. 

It is a bit cumbersome having to install with `pip install .` every time I want to test a change. I could create sylinks to the file inside the `site-packages/myproject`. Is this a good practice? Is there other workaround?

**Comment by abravalheri** [comment, marked as answer]
Hi @juanmaia, have you tried to use `pip install -e .`? This `-e` means that you install the project in the "editable mode". I believe this way the changes will be always up to date.

The only thing that you have to pay attention is that, if you add/remove dependencies, entry-points, or more generally modify the `setup.cfg` file, you will have to run the `pip install -e .` again (because setuptools will have to re-generate the metadata).

**Reply by juanmaia** [reply]
Thanks @abravalheri this works just fine. I assume any other solution without using `pip install` would be considered a hack, right? Like messing with sys.path or something similar.

**Reply by abravalheri** [reply]
Yes, I am afraid so 😅.

Messing with `sys.path` or "cd-ing" into the `src` directory and running the commands from there is something you could do... But these alternative approaches have their own drawbacks (e.g. dependencies and entry-points are not installed, and the metadata - which include package name and version - is not generated).

**Reply by FlorianWilhelm** [reply]
Hi @juanmaia, would you mind marking this question as answered? You could choose for instance @abravalheri answer and select it.

**Reply by juanmaia** [reply]
@FlorianWilhelm it seem that it was previously marked as "Best practices" which was not an "answerable" category.  Done.

**Reply by FlorianWilhelm** [reply]
yes, I had to move them to the discussion category Q&A as your are asking questions. Now it should show up.
